metrics/internal/collectors/rbd-children-count.go

```diff
@@ -0,0 +1,70 @@
+package collectors
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	internalcache "github.com/red-hat-storage/ocs-operator/metrics/v4/internal/cache"
+	"github.com/red-hat-storage/ocs-operator/metrics/v4/internal/options"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+var _ prometheus.Collector = &CephRBDChildrenCollector{}
+
+type CephRBDChildrenCollector struct {
+	kubeClient               clientset.Interface
+	PersistentVolumeStore    *internalcache.PersistentVolumeStore
+	StorageConsumerCollector *StorageConsumerCollector
+	// Metrics Descriptors
+	RBDChildrenCount *prometheus.Desc
+}
+
+func NewCephRBDChildrenCollector(pvStore *internalcache.PersistentVolumeStore, opts *options.Options, storageConsumerCollector *StorageConsumerCollector) *CephRBDChildrenCollector {
+	return &CephRBDChildrenCollector{
+		kubeClient:               clientset.NewForConfigOrDie(opts.Kubeconfig),
+		PersistentVolumeStore:    pvStore,
+		StorageConsumerCollector: storageConsumerCollector,
+		RBDChildrenCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "rbd", "children_count"),
+			"RBD children count", []string{"image", "radosnamespace", "consumerid"}, nil),
+	}
+}
+
+func (c *CephRBDChildrenCollector) Describe(ch chan<- *prometheus.Desc) {
+	ds := []*prometheus.Desc{
+		c.RBDChildrenCount,
+	}
+
+	for _, d := range ds {
+		ch <- d
+	}
+}
+
+func (c *CephRBDChildrenCollector) Collect(ch chan<- prometheus.Metric) {
+	c.PersistentVolumeStore.Mutex.RLock()
+	defer c.PersistentVolumeStore.Mutex.RUnlock()
+
+	var consumerId string
+	if c.StorageConsumerCollector != nil {
+		consumerId = c.StorageConsumerCollector.GetClientId()
+		if consumerId == "" {
+			klog.Infof("consumerId is empty")
+		}
+	}
+
+	for _, attrs := range c.PersistentVolumeStore.Store {
+		image := attrs.ImageName
+		radosNamespace := attrs.RadosNameSpace
+
+		count, ok := c.PersistentVolumeStore.RBDChildrenMap[image]
+		if !ok {
+			continue // skip if no children count
+		}
+
+		ch <- prometheus.MustNewConstMetric(
+			c.RBDChildrenCount,
+			prometheus.GaugeValue,
+			float64(count),
```

Comment:
Is there a reason why we use float64 for this? Doesn't prom client allow a time series that is consistent of integers (which better represent a count)